Code/GraphMol/Substruct/SubstructMatch.cpp

```diff
@@ -439,58 +458,50 @@ std::vector<MatchVectType> SubstructMatch(
   if (!mol.getNumAtoms() || !query.getNumAtoms()) {
     return matches;
   }
-  std::vector<RecursiveStructureQuery *> locked;
-  locked.reserve(query.getNumAtoms());
+
+  detail::RecursiveLocker locker(query, params.recursionPossible);
+
   if (params.recursionPossible) {
     detail::SUBQUERY_MAP subqueryMap;
     ROMol::ConstAtomIterator atIt;
     for (atIt = query.beginAtoms(); atIt != query.endAtoms(); atIt++) {
       if ((*atIt)->getQuery()) {
-        // std::cerr<<"recurse from atom "<<(*atIt)->getIdx()<<std::endl;
-        detail::MatchSubqueries(mol, (*atIt)->getQuery(), params, subqueryMap,
-                                locked);
+	// std::cerr<<"recurse from atom "<<(*atIt)->getIdx()<<std::endl;
+	detail::MatchSubqueries(mol, (*atIt)->getQuery(), params, subqueryMap,
+				locker.locked);
```

Comment:
I guess your code formatter did this automatically, but it's not correct: these two lines should be indented as they were before